subliminal/providers/legendastv.py

```diff
@@ -0,0 +1,466 @@
+            data = {'_method': 'POST', 'data[User][username]': self.username, 'data[User][password]': self.password}
+            r = self.session.post('%s/login' % self.server_url, data, allow_redirects=False, timeout=TIMEOUT)
+            r.raise_for_status()
+
+            soup = ParserBeautifulSoup(r.content, ['lxml', 'html.parser'])
+            auth_error = soup.find('div', {'class': 'alert-error'}, text=re.compile(u'.*Usuário ou senha inválidos.*'))
+
+            if auth_error:
+                raise AuthenticationError(self.username)
+
+            logger.debug('Logged in')
+            self.logged_in = True
+
+    def terminate(self):
+        # logout
+        if self.logged_in:
+            logger.info('Logging out')
+            r = self.session.get('%s/users/logout' % self.server_url, timeout=TIMEOUT)
+            r.raise_for_status()
+            logger.debug('Logged out')
+            self.logged_in = False
+
+        self.session.close()
+
+    def matches(self, expected, actual, ignore_episode=False):
+        """
+        Matches two dictionaries (expected and actual). The dictionary keys follow the guessit properties names.
+        If the expected dictionary represents a movie:
+          - ``type`` should match
+          - ``title`` should match
+          - ``year`` should match, unless they're not defined and expected and actual ``title``s are the same
+        If the expected dictionary represents an episode:
+          - ``type`` should match
+          - ``series`` should match
+          - ``eason`` should match
+          - ``episodeNumber`` should match, unless ``ignore_episode`` is True
+
+        :param expected: dictionary that contains the expected values
+        :param actual: dictionary that contains the actual values
+        :param ignore_episode: True if should ignore episodeNumber matching. Default: False
+        :return: True if actual matches expected
+        :rtype: bool
+        """
+        if expected.get('type') != actual.get('type'):
+            return False
+
+        if expected.get('type') == 'movie':
+            if not self.name_matches(expected.get('title'), actual.get('title')):
+                return False
+            if expected.get('year') != actual.get('year'):
+                if expected.get('year') and actual.get('year'):
+                    return False
+                if expected.get('title', '').lower() != actual.get('title', '').lower():
+                    return False
+
+        elif expected.get('type') == 'episode':
+            if not self.name_matches(expected.get('series'), actual.get('series')):
+                return False
+            if expected.get('season') != actual.get('season'):
+                return False
+            if not ignore_episode and expected.get('episodeNumber') != actual.get('episodeNumber'):
+                return False
+
+        return True
+
+    def name_matches(self, expected_name, actual_name):
```

Comment:
I meant can you get rid of `name_matches` and use `sanitized_string_equal` instead? For your examples you can just `s/[SE]\d{2}//g` and send the result to `sanitized_string_equal`. I'd prefer have only one function that does the string match, if need be it can be improved to fit your use case.
I may need to add parameterized unittests to `sanitized_string_equal`